crates/bevy_ecs/src/change_detection.rs

```diff
@@ -932,4 +963,40 @@ mod tests {
             "Resource must be changed after setting to a different value."
         );
     }
+
+    #[test]
+    fn mut_untyped_to_reflect() {
+        let (last_change_tick, change_tick) = (2, 3);
+        let mut component_ticks = ComponentTicks {
+            added: Tick::new(1),
+            changed: Tick::new(2),
+        };
+        let ticks = TicksMut {
+            added: &mut component_ticks.added,
+            changed: &mut component_ticks.changed,
+            last_change_tick,
+            change_tick,
+        };
+
+        let mut value: i32 = 5;
+        let value = MutUntyped {
+            // SAFETY: lifetime does not exceed `value`
+            value: unsafe { PtrMut::new(NonNull::new(&mut value as *mut i32 as *mut u8).unwrap()) },
+            ticks,
+        };
```

Comment:
```suggestion
        let value = MutUntyped {
            value: PtrMut::from(&mut value),
            ticks,
        };
```

One more thing I just noticed: we don't need `unsafe` here.